File: python_codes/readability.py
# Get the text for input
text = input("text: ")

# Count the number of letters
count_letter = 0
for letter in text:
    if (letter.isalpha()):
        count_letter += 1
# Count the number of words
words = len(text.split())

# Count the number of sentences
text = text.replace('!', '.')
text = text.replace('?', '.')
sentences = len(text.split('.')) - 1

# Tests all parameters
# print(count_letter)
# print(words)
# print(sentences)

# Coleman-Liau
L = float((count_letter / words) * 100)
S = float((sentences / words) * 100)
index = int(round(0.0588 * L - 0.296 * S - 15.8))
# Test the index
# print(index)

if (index < 1):
    print("Before Grade 1")
elif (index >= 16):
    print("Grade 16+")
else:
    print(f"Grade {index}")

File: readability/readability.c
#include <math.h>
#include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <string.h>

int main(void)
{
    string text = get_string("Text: ");

    //Getting the number of letters in the text.
    int letters = 0;
    for (int i = 0; text[i] != '\0'; i++)
    {
        if (text[i] >= 'A' && text[i] <= 'Z')
        {
            letters++;
        }
        else if (text[i] >= 'a' && text[i] <= 'z')
        {
            letters++;
        }
        else
        {

        }
    }
    //Test number of letters
    //printf("%i letter(s)\n", letters);

    //Getting the number of words in the text.
    int words = 0;
    for (int i = 0; text[i] != '\0'; i++)
    {
        if (text[i] >= 'A' && text[i] <= 'Z')
        {

        }
        else if (text[i] >= 'a' && text[i] <= 'z')
        {

        }
        else if (text[i] == 33 || text[i] == 34 || text[i] == 39 || text[i] == 44 || text[i] == 46 || text[i] == 45 || text[i] == 58
                 || text[i] == 59 || text[i] == 96)
        {

        }
        else
        {
            words++;

        }
    }
    int words2 = 0;
    words2 = words + 1;
    //Testing the number of words
    //printf("%i word(s)\n", words2);

    //Getting the number of sentences in the text.
    int sentences = 0;
    for (int i = 0; text[i] != '\0'; i++)
    {
        if (text[i] == 33 || text[i] == 46 || text[i] == 63)
        {
            sentences++;
        }
        else
        {

        }
    }
    //Testing the number of sentences
    //printf("%i sentence(s)\n", sentences);

    //Putting it All Together
    float L = (letters / (float)(words2)) * 100;
    float S = (sentences / (float)(words2)) * 100;
    int index = round(0.0588 * L - 0.296 * S - 15.8);
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}